engine.py

```diff
@@ -443,29 +445,26 @@ def main(args):
     """
     Kicks off the engine with the given arguments.
     """
-    try:
-        nat_config = {}
-        if args.get("sender_ip") and args.get("routing_ip") and args.get("forward_ip"):
-            nat_config = {"sender_ip" : args["sender_ip"],
-                          "routing_ip" : args["routing_ip"],
-                          "forward_ip" : args["forward_ip"]}
-
-        eng = Engine(args["server_port"],
-                     args["strategy"],
-                     environment_id=args["environment_id"],
-                     server_side=args["server_side"],
-                     output_directory=args["output_directory"],
-                     forwarder=nat_config,
-                     log_level=args["log"],
-                     in_queue_num=args["in_queue_num"],
-                     out_queue_num=args["out_queue_num"],
-                     save_seen_packets=args["no_save_packets"],
-                     demo_mode=args["demo_mode"])
+    nat_config = {}
+    if args.get("sender_ip") and args.get("routing_ip") and args.get("forward_ip"):
+        nat_config = {"sender_ip": args["sender_ip"],
+                      "routing_ip": args["routing_ip"],
+                      "forward_ip": args["forward_ip"]}
+
+    with Engine(args["server_port"],
+                args["strategy"],
+                environment_id=args["environment_id"],
+                server_side=args["server_side"],
+                output_directory=args["output_directory"],
+                forwarder=nat_config,
+                log_level=args["log"],
+                in_queue_num=args["in_queue_num"],
+                out_queue_num=args["out_queue_num"],
+                save_seen_packets=args["no_save_packets"],
+                demo_mode=args["demo_mode"]) as eng:
         eng.initialize_nfqueue()
-        while True:
-            time.sleep(0.5)
-    finally:
-        eng.shutdown_nfqueue()
```

Comment:
i think we might still want the finally to handle ^C or other catchable exceptions when the engine is invoked directly

Reply:
The \_\_exit\_\_-method of Engine is always called - even in case of an exception: https://book.pythontips.com/en/latest/context_managers.html#handling-exceptions

Since shutdown_nfqueue is called from there, we can (and should) remove the finally-block in my opinion.

Reply:
Ah you're right - I was remembering that this code did not use the context manager, and had missed that you swapped it to using it. Thanks for doing it - I'll merge it!